gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTest.java

```diff
@@ -1247,6 +1247,141 @@ public void testUpdateItemInfoUpdatesMetadata() throws IOException {
+      GoogleCloudStorageItemInfo itemInfo = itemInfos.get(i);
+
+      assertThat(itemInfo.getResourceId()).isEqualTo(resourceId);
+      assertThat(itemInfo.getSize()).isEqualTo(0);
+      assertThat(itemInfo.exists()).isTrue();
+      assertThat(itemInfo.getCreationTime()).isNotEqualTo(0);
+      assertThat(itemInfo.isBucket()).isFalse();
+      assertThat(itemInfo.isDirectory()).isFalse();
+    }
+  }
+
+  @Test
+  public void testListObjectInfoBeforeAndAfterCreate() throws IOException {
+    String bucketName = getSharedBucketName();
+    String objectName = "testListObjectInfoBeforeAndAfterCreate_";
+
+    List<StorageResourceId> resourceIds =
+        ImmutableList.of(
+            new StorageResourceId(bucketName, objectName + "x"),
+            new StorageResourceId(bucketName, objectName + "y"),
+            new StorageResourceId(bucketName, objectName + "z"));
+
+    assertThat(rawStorage.listObjectInfo(bucketName, objectName, "/")).isEmpty();
+
+    rawStorage.createEmptyObjects(resourceIds);
+
+    assertThat(rawStorage.listObjectInfo(bucketName, objectName, "/")).hasSize(3);
+  }
+
+  @Test
+  public void testlistObjectInfoPageBeforeAndAfterCreate() throws IOException {
+    String bucketName = getSharedBucketName();
+    String objectName = "testlistObjectInfoPageBeforeAndAfterCreate_Object";
+
+    StorageResourceId resourceId = new StorageResourceId(bucketName, objectName);
+
+    ListPage<GoogleCloudStorageItemInfo> itemInfosPage =
+        rawStorage.listObjectInfoPage(bucketName, objectName, "/", /* pageToken= */ null);
+
+    assertThat(itemInfosPage.getNextPageToken()).isNull();
+    assertThat(itemInfosPage.getItems()).isEmpty();
+
+    writeObject(rawStorage, resourceId, /* objectSize= */ 512);
+
+    itemInfosPage =
+        rawStorage.listObjectInfoPage(bucketName, objectName, "/", /* pageToken= */ null);
+
+    assertThat(itemInfosPage.getNextPageToken()).isNull();
+    assertThat(itemInfosPage.getItems()).hasSize(1);
+    assertThat(itemInfosPage.getItems().get(0).getResourceId()).isEqualTo(resourceId);
+  }
+
+  @Test
+  public void testOverwriteExistingObject() throws IOException {
+    String bucketName = getSharedBucketName();
+
+    StorageResourceId resourceId =
+        new StorageResourceId(bucketName, "testOverwriteExistingObject_Object");
+
+    assertThat(rawStorage.getItemInfo(resourceId).exists()).isFalse();
+
+    writeObject(rawStorage, resourceId, /* objectSize= */ 128);
+
+    GoogleCloudStorageItemInfo createdItemInfo = rawStorage.getItemInfo(resourceId);
+    assertThat(createdItemInfo.exists()).isTrue();
+    assertThat(createdItemInfo.getSize()).isEqualTo(128);
+
+    byte[] objectContent = writeObject(rawStorage, resourceId, /* objectSize= */ 256);
+
+    GoogleCloudStorageItemInfo overwrittenItemInfo = rawStorage.getItemInfo(resourceId);
+    assertThat(overwrittenItemInfo.exists()).isTrue();
+    assertThat(overwrittenItemInfo.getSize()).isEqualTo(256);
```

Comment:
I have improved tests a little and seems found a bug. This line is failing with perf cache. May you take a look into it?